.github/workflows/deploy-gcp-tests.yml

```diff
@@ -241,24 +241,18 @@ jobs:
 
           # Try to find an image generated from the main branch
           # Fields are listed in the "Create image from state disk" step
-          #
-          # TODO: require ${NETWORK} in the name after PR #4391 merges to main, and runs a full sync
-          #       network should replace [a-z]*
-          CACHED_DISK_NAME=$(gcloud compute images list --filter="name~${DISK_PREFIX}-main-[0-9a-f]+-v${LOCAL_STATE_VERSION}-[a-z]*-${{ inputs.disk_suffix }}" --format="value(NAME)" --sort-by=~creationTimestamp --limit=1)
+          CACHED_DISK_NAME=$(gcloud compute images list --filter="name~${DISK_PREFIX}-main-[0-9a-f]+-v${LOCAL_STATE_VERSION}-${NETWORK}-${{ inputs.disk_suffix }}" --format="value(NAME)" --sort-by=~creationTimestamp --limit=1)
```

Comment:
This code is partly duplicated in `get-available-disks` in PR #4385, so this PR also needs to add the network to that file.

Reply:
Done